base/error.jl

```diff
@@ -49,3 +49,50 @@ macro assert(ex, msgs...)
     end
     :($(esc(ex)) ? $(nothing) : throw(Main.Base.AssertionError($msg)))
 end
+
+
+"""
+    retry(f, [condition]; n=3; max_delay=10) -> Function
+
+Returns a lambda that retries function `f` up to `n` times in the
+event of an exception. If `condition` is a `Type` then retry only
+for exceptions of that type. If `condition` is a function
+`cond(::Exception) -> Bool` then retry only if it is true.
+
+e.g. `retry(http_get, e->e.status == "503")(url)` or `retry(read, UVError)(io)`.
+"""
+function retry(f::Function, condition::Function=e->true;
+               n::Int=3, max_delay::Int=10)
+
+    (args...) -> begin
+        delay = 0.05
+        for i = 1:n
+            try
+                return f(args...)
+            catch e
+                if i == n || try condition(e) end != true
+                    rethrow(e)
+                end
+            end
+            sleep(delay * (0.8 + (rand() * 0.4)))
+            delay = min(max_delay, delay * 5)
+        end
+    end
+end
+
+retry(f::Function, t::Type; kw...) = retry(f, e->isa(e, t); kw...)
+
+
+"""
+    @catch(f) -> Function
+
+Returns a lambda that executes `f` and returns either the result of `f` or
+an `Exception` thrown by `f`.
+
+e.g. `@catch(length)() == MethodError(length,())`
```

Comment:
Same here. See the new guidelines at http://docs.julialang.org/en/latest/manual/documentation/